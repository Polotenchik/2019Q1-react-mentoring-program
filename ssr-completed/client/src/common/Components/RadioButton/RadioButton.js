import React, { Fragment } from 'react';
import styled, { css } from 'styled-components';
import Theme from '../../Theme/Theme';

const RadioButton = ({
  inputName,
  inputValue,
  inputId,
  filtered,
  handlerChange,
  ...props
}) => (
  <Fragment>
    <RadioInput
      type="radio"
      name={inputName}
      value={inputValue}
      id={inputId}
      onChange={(evt) => handlerChange(evt.target.id, evt.target.value)} />
    <RadioLabel
      htmlFor={inputId}
      id={filtered === inputValue ? 'active' : undefined}
      active={filtered === inputId}
      {...props}>
      {inputValue}
    </RadioLabel>
  </Fragment>
);

export default RadioButton;

const RadioInput = styled.input`
    display: none;
`;

const RadioLabel = styled.label`
  cursor: pointer;
  margin-left: 20px;    
  border-radius: 4px;
  cursor: pointer;

   ${props => props.underline && props.active && css`
    color: ${Theme.colors.red};
    text-decoration: underline;
   `}
   
   ${props => props.colored && css`
        background-color: ${Theme.colors.medium_grey};
        color:  ${Theme.colors.white};   
        text-transform: uppercase;   
        padding: 6px 12px;         
   `}   
   
    ${props => props.colored && props.active && css`
        background-color: ${Theme.colors.red}
     `} 
   
    ${props => props.transparent && css`
        background-color: transparent;         
   `}
   
    ${props => props.transparent && props.active && css`
        color: ${Theme.colors.red}
    `} 
`;
