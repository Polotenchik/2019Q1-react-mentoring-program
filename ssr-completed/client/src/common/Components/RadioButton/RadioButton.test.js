import React from 'react';
import { shallow } from 'enzyme';
import RadioButton from './RadioButton';

describe('RadioButton Snapshot', () => {
    it('RadioButton Component should render correctly with no props', () => {
        const component = shallow(<RadioButton />);
        expect(component).toMatchSnapshot();
    });

    it('RadioButton Component should render radio button correctly with given props', () => {
        const inputName = 'test_button';
        const inputValue = 'radio button';
        const inputId = 'radioButton';
        const filtered = 'radio button';
        const handlerChange = jest.fn();
        const component = shallow(<RadioButton inputName={inputName} inputValue={inputValue} inputId={inputId} filtered={filtered} handlerChange={handlerChange}/>);
        expect(component).toMatchSnapshot();
    });
});

