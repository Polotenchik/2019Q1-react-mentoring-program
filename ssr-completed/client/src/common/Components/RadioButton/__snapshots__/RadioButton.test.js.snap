// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Snapshot RadioButton Component should render correctly with no props 1`] = `
<Fragment>
  <styled.input
    onChange={[Function]}
    type="radio"
  />
  <styled.label
    active={true}
    id="active"
  />
</Fragment>
`;

exports[`RadioButton Snapshot RadioButton Component should render radio button correctly with given props 1`] = `
<Fragment>
  <styled.input
    id="radioButton"
    name="test_button"
    onChange={[Function]}
    type="radio"
    value="radio button"
  />
  <styled.label
    active={false}
    htmlFor="radioButton"
    id="active"
  >
    radio button
  </styled.label>
</Fragment>
`;
